tasks/plugins.yml

```diff
@@ -45,14 +45,20 @@
     path: "{{ jenkins_home }}/updates/default.json"
     regexp: "1d;$d"
 
+# Added ability to specify version of a plugin.
+# jenkins_plugins:
+#   - blueocean
+#   - { name: "influxdb", version: "1.12.1" }
+
 - name: Install Jenkins plugins using password.
   jenkins_plugin:
-    name: "{{ item }}"
+    name: "{{ item.name | default(item) }}"
+    version: "{{ item.version | default(omit) }}"
     jenkins_home: "{{ jenkins_home }}"
     params:
       url_username: "{{ jenkins_admin_username }}"
     url_password: "{{ jenkins_admin_password }}"
-    state: "{{ jenkins_plugins_state }}"
+    state: "{{ 'present' if item.version is defined else jenkins_plugins_state }}"
```

Comment:
I'm not sure why this should be overridden here. I can understand if maybe `item.present` were an allowed value, falling back to `jenkins_plugins_state`, but I don't know why it should be present if a version is set (that doesn't seem to follow, logically).

Reply:
Thinking is (at least in my configuration) that `jenkins_plugins_state` is globally set to `latest`. For the majority of plugins, we want to update to the latest version. However, the version is pinned than having this parameter be latest does not make sense. We could omit it or pass in `present`.
I am not sure what `jenkins_plugin` behavior is when we pass in specific version and state latest together.

Reply:
Having said that would you prefer omit or something else?

Reply:
I see you point. Maybe add in the docs in the README for `jenkins_plugins_state` something like "will be overridden and set to `present` for any plugin that has a specific `version` set".

Reply:
👍